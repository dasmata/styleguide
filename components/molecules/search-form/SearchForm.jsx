import React from 'react';
import PropTypes from 'prop-types';

import Input from '../../atoms/input/Input';
import Button from '../../atoms/button/Button';

const prefix = 'm-search';
const MIN_VALUE = 3;

class SearchForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      results: [],
    };
  }

  handleSearchInput(value) {
    const { onInput } = this.props;
    if (value.length >= MIN_VALUE && onInput) {
      onInput(value);
    }
  }

  render() {
    const { results } = this.state;
    const { input, button } = this.props;
    return (
      <div className={prefix}>
        <Input
          type="text"
          onChange={((value) => this.handleSearchInput(value))}
          {...input}
        />
        <Button {...button} className={`${prefix}__button`} />
        <div>
          {results.map((el) => (<div>el</div>))}
        </div>
      </div>
    );
  }
}

SearchForm.propTypes = {
  button: PropTypes.shape({
    text: PropTypes.string
  }),
  input: PropTypes.shape({
    placeholder: PropTypes.string
  }),
  onInput: PropTypes.func
};

export default SearchForm;
