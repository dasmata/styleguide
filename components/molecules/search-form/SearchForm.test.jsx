import React from 'react';
import { shallow } from 'enzyme';

import searchConfig from './data/00-header.json';
import SearchForm from './SearchForm';

describe('Input atom snapshot tests', () => {
  it('renders text input [no errors]', () => {
    const tree = shallow(<SearchForm {...searchConfig.moduleConfig} />);

    expect(tree).toMatchSnapshot();
  });
});
