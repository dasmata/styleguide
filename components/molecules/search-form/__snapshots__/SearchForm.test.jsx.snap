// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input atom snapshot tests renders text input [no errors] 1`] = `
<div
  className="m-search"
>
  <Input
    onChange={[Function]}
    placeholder="Search everything!!!"
    type="text"
  />
  <Button
    text="Fetch!"
  />
  <div />
</div>
`;
